test/RandomTest.cpp

```diff
@@ -235,4 +236,23 @@ TEST(RandomShuffleTest, Seed) {
       });
 }
 
+TEST(UuidGeneratorTest, StrUuidGeneratorTest) {
+  // Test few times that the returned UUID str is not
+  // "00000000-0000-0000-0000-000000000000" (nil-UUID)
+  // and that none of the str-UUIDS is rquivalent to the already
+  // created ones.
```

Comment:
You could also add a test, that the UUIDs follow a certain schema/ fulfill a certain regex (you know the exact format better than I do).